src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -755,6 +756,41 @@ public StudyEntity doRenameStudy(UUID studyUuid, String userId, String newStudyN
                 .map(StudyService::toStudyInfos);
     }
 
+    private Mono<Void> applyLineChanges(UUID studyUuid, String path, String status) {
+        Mono<Void> monoUpdateLfState = updateLoadFlowResultAndStatus(studyUuid, null, LoadFlowStatus.NOT_DONE)
+                .doOnSuccess(e -> emitStudyChanged(studyUuid, UPDATE_TYPE_LOADFLOW_STATUS))
+                .then(invalidateSecurityAnalysisStatus(studyUuid)
+                        .doOnSuccess(e -> emitStudyChanged(studyUuid, UPDATE_TYPE_SECURITY_ANALYSIS_STATUS)))
+                .doOnSuccess(e -> emitStudyChanged(studyUuid, UPDATE_TYPE_LINE));
+        Flux<ElementaryModificationInfos> fluxChangeLineStatus = webClient.put()
+                .uri(networkModificationServerBaseUri + path)
+                .body(BodyInserters.fromValue(status))
+                .retrieve()
+                .onStatus(httpStatus -> httpStatus == HttpStatus.INTERNAL_SERVER_ERROR, clientResponse -> Mono.error(new StudyException(LINE_MODIFICATION_FAILED)))
```

Comment:
Original error message is now returned in response body. Did not add a specific treatment in case line is not found, the error message in this case will be LINE_NOT_FOUND.